File: models/Note.js
// Create a new model for the Note table.
const { Model, DataTypes } = require('sequelize');
// Import the connection to the database.
const sequelize = require('../config/connection');
// This is the Note model that extends the Sequelize Model.
class Note extends Model {}

// Define the columns in the Note table.
Note.init(
  {
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true,
      autoIncrement: true,
    },
    title: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    content: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    User_id: {
      type: DataTypes.INTEGER,
      references: {
        model: 'User',
        key: 'id',
      },
    },
  },
  {
    sequelize,
    freezeTableName: true,
    underscored: true,
    timestamps: true,
    modelName: 'Note',
  }
);

module.exports = Note;

File: models/Note.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app import db  # Import db from app.py

class Note(db.Model):
    __tablename__ = 'note'
    id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String, nullable=False)
    content = Column(String, nullable=False)
    user_id = Column(Integer, ForeignKey('user.id'), nullable=False)
    
    user = relationship('User', back_populates='notes')
    
    def __repr__(self):
        return f'<Note {self.title}>'

